src/Cli/dotnet/Telemetry/DockerContainerDetectorForTelemetry.cs

```diff
@@ -48,9 +48,20 @@ public IsDockerContainer IsDockerContainer()
 
         private static bool ReadProcToDetectDockerInLinux()
         {
-            return File
-                .ReadAllText("/proc/1/cgroup")
-                .Contains("/docker/");
+            try
+            {
+                return File
+                    .ReadAllText("/proc/1/cgroup")
+                    .Contains("/docker/");
+            }
+            catch (Exception ex) when (ex is IOException || ex.InnerException is IOException)
+            {
+                // in some environments (restricted docker container, shared hosting etc.),
+                // procfs is not accessible and we get UnauthorizedAccessException while the
+                // inner exception is set to IOException. Ignore and continue when that happens.
```

Comment:
It sounds like if we get an exception here, we don't know if we're in a docker container or not.  So should we return `IsDockerContainer.Unkown` from `IsDockerContainer` in that case?

Reply:
Sure, updated. I didn't realize that there exist `IsDockerContainer.Unknown`. :)